### PR TITLE
Update UI for new reference panel

### PR DESCRIPTION
**Pull request by Numbers88s**
Ref #35725, #35721, #35723, #35726 

Update UI based on [figma file](https://www.figma.com/file/DZMyEJT3g2HT6HmJGNNOnW/%22Better%22-code-intel-panel?node-id=1384%3A10928).

New Reference Panel
CleanShot 2022-05-24 at 21 49 03

## Test plan
Manually tested. Most changes are visual

## App preview:

- [Web](https://sg-web-cesar-update-ui-in-reference-panel.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-sazbrfsqzi.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/codeintel/ReferencesPanel.test.tsx [hunk 1]
```diff
@@ -158,9 +158,6 @@ describe('ReferencesPanel', () => {
         const closeCodeViewButton = result.getByRole('button', { name: 'Close code view' })
         expect(closeCodeViewButton).toBeVisible()
 
-        const fileLink = result.getByRole('link', { name: 'diff/diff.go' })
-        expect(fileLink).toBeVisible()
-
```
Comment (mrnugget) [inline client/web/src/codeintel/ReferencesPanel.test.tsx:163]: Why remove that assertion? This tests that the header of the "peek" code view on the right exists (`<SideBlob />`). That's still unchanged, no? Do we need a `const fileLink = within(codeview).getByRole` here?

**Review by mrnugget — APPROVED** [review]
Nice! Thank you for doing this! This is immensely helpful. I have two comments that I think you should look at: the `overflow: scroll` thing and the removed test assertion.

**Comment by mrnugget** [comment]
This one's more for @rrhyne so I'm leaving it as a comment: the padding of the list looks a bit weird when the first or last entry are active
screenshot_2022-05-25_09 34 05@2x

File: client/web/src/codeintel/ReferencesPanel.module.scss [hunk 2]
```diff
@@ -108,13 +134,34 @@ $group-divider-height: 0.25rem;
...
 
+.location-container {
+    padding-left: 3.9rem;
+    padding-right: 1rem;
+
+    ul {
+        border: 1px solid var(--gray-03);
+        padding: 9px 0;
+        border-radius: 3px;
+        overflow: scroll;
```
[the remark this answers is not in the document]
Comment (Numbers88s) [reply client/web/src/codeintel/ReferencesPanel.module.scss:160]: I just thought it needed it. Happy to remove it.

**Comment by Numbers88s** [comment]
> This one's more for @rrhyne so I'm leaving it as a comment: the padding of the list looks a bit weird when the first or last entry are active screenshot_2022-05-25_09 34 05@2x

Agreed. I have a fix for it where it highlights the whole box (when there is only one) but then it looks weird when there is a list since its a thin line. So I thought to be consistent it should be a thin line for it all. Let me know what you think.

I do not have a fix for the last and first of the list tho.

File: client/web/src/codeintel/ReferencesPanel.test.tsx (hunk 1 above)
Comment (Numbers88s) [reply client/web/src/codeintel/ReferencesPanel.test.tsx:163]: Sorry about that. I misunderstood what the test did. Added it back

File: client/web/src/codeintel/ReferencesPanel.tsx [hunk 3]
```diff
@@ -369,30 +369,12 @@ export const ReferencesList: React.FunctionComponent<
...
-                    )}
-                </CardHeader>
-                <div className={classNames('d-flex justify-content-start', styles.filter)}>
+                <div className={classNames('d-flex justify-content-start mt-2', styles.filter)}>
                     <small>
                         <Icon
                             role="img"
                             aria-hidden={true}
-                            as={FilterOutlineIcon}
+                            as={canShowSpinner ? LoadingSpinner : FilterOutlineIcon}
```
[the remark this answers is not in the document]
Comment (Numbers88s) [reply client/web/src/codeintel/ReferencesPanel.tsx:377]: awesome 🤩

**Pull request merged**